### ISSUE TITLE
Implement Priority Replay Buffer

### ISSUE THREAD

**Issue by RichardDev01** [HU-ICT-LAB/RobotWars#145]
# Description

As developers, we need to develop our replay buffer with functionality that sorts it based on the loss of each training sample.

# Why do we need to do this?

This story is needed to train on a prioritised training set for better performance whilst training

# Tasks

- [ ] Implement the Prioritised Replay Buffer
- [ ] Test the Prioritised Replay buffer

# Acceptance Criteria

This user story is done when the Prioritised Replay Buffer has been implemented in the pipeline and has been tested.
